### PR TITLE
Fix packageVersion so it can be appended before rc tag

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/30496

**Comment by StrongMonkey** [comment]
```
Successfully packaged chart and saved it to: assets/fleet/fleet-0.3.200-rc5.tgz
Successfully packaged chart and saved it to: assets/fleet-agent/fleet-agent-0.3.200-rc5.tgz
Successfully packaged chart and saved it to: assets/fleet-crd/fleet-crd-0.3.200-rc5.tgz
Successfully packaged chart and saved it to: assets/longhorn/longhorn-1.0.202.tgz
Successfully packaged chart and saved it to: assets/longhorn/longhorn-crd-1.0.202.tgz
Successfully packaged chart and saved it to: assets/rancher-backup/rancher-backup-1.0.300.tgz
Successfully packaged chart and saved it to: assets/rancher-backup/rancher-backup-crd-1.0.300.tgz
Successfully packaged chart and saved it to: assets/rancher-cis-benchmark/rancher-cis-benchmark-1.0.300.tgz
Successfully packaged chart and saved it to: assets/rancher-cis-benchmark/rancher-cis-benchmark-crd-1.0.300.tgz
Successfully packaged chart and saved it to: assets/rancher-external-ip-webhook/rancher-external-ip-webhook-0.1.600.tgz
Successfully packaged chart and saved it to: assets/rancher-gatekeeper/rancher-gatekeeper-3.2.101.tgz
Successfully packaged chart and saved it to: assets/rancher-gatekeeper/rancher-gatekeeper-crd-3.2.101.tgz
Successfully packaged chart and saved it to: assets/rancher-istio/rancher-istio-1.7.302.tgz
Successfully packaged chart and saved it to: assets/rancher-kiali-server/rancher-kiali-server-1.24.003.tgz
Successfully packaged chart and saved it to: assets/rancher-kiali-server/rancher-kiali-server-crd-1.24.003.tgz
Successfully packaged chart and saved it to: assets/rancher-logging/rancher-logging-3.8.201.tgz
Successfully packaged chart and saved it to: assets/rancher-logging/rancher-logging-crd-3.8.201.tgz
Successfully packaged chart and saved it to: assets/rancher-monitoring/rancher-monitoring-9.4.202.tgz
Successfully packaged chart and saved it to: assets/rancher-monitoring/rancher-monitoring-crd-9.4.202.tgz
Successfully packaged chart and saved it to: assets/rancher-operator/rancher-operator-0.1.200-rc2.tgz
Successfully packaged chart and saved it to: assets/rancher-operator-crd/rancher-operator-crd-0.1.200-rc2.tgz
Successfully packaged chart and saved it to: assets/rancher-pushprox/rancher-pushprox-0.1.2.tgz
Successfully packaged chart and saved it to: assets/rancher-tracing/rancher-tracing-1.20.001.tgz
Successfully packaged chart and saved it to: assets/rancher-webhook/rancher-webhook-0.1.0-beta900.tgz
Successfully packaged chart and saved it to: assets/rio/rio-0.8.000.tgz
```

File: scripts/generate-charts [hunk 1]
```diff
@@ -11,7 +25,7 @@ for f in packages/*; do
 		if [[ -f ${f}/package.yaml ]]; then
 			version=$(yq r ${f}/charts/Chart.yaml version)
 			packageVersion=$(yq r ${f}/package.yaml packageVersion)
-			yq w -i ${f}/charts/Chart.yaml 'version' "${version}${packageVersion}"
+			yq w -i ${f}/charts/Chart.yaml 'version' "$(packageVersion ${version} ${packageVersion})"
```
Comment (aiyengar2) [inline scripts/generate-charts:28]: can we rename the function to something else? Seems confusing here

File: packages/fleet-agent/package.yaml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 url:  https://github.com/rancher/fleet/releases/download/v0.3.2-rc5/fleet-agent-0.3.2-rc5.tgz
-packageVersion: 00
+packageVersion: 00
```
Comment (aiyengar2) [inline packages/fleet-agent/package.yaml:2]: Any reason why we need the spaces?

Comment (aiyengar2) [reply packages/fleet-agent/package.yaml:2]: nvm, need it to regenerate assets.

**Review by aiyengar2 — APPROVED** [review]
LGTM

**Pull request merged**